### PR TITLE
make FakeKernel#exit behave like real Kernel#exit

### PR DESCRIPTION
**Pull request by grosser**
fixes https://github.com/cucumber/aruba/issues/592

@olleolleolle

### CODE REVIEW

File: lib/aruba/processes/in_process.rb [hunk 1]
```diff
@@ -33,7 +33,12 @@ def initialize
         end
 
         def exit(exitstatus)
-          @exitstatus = exitstatus
+          @exitstatus =
+            case exitstatus
+            when Numeric then Integer(exitstatus)
+            when TrueClass, FalseClass then exitstatus ? 0 : 1
+            else raise TypeError, "no implicit conversion of #{exitstatus.class} into Integer"
```
Comment (mvz) [inline lib/aruba/processes/in_process.rb]: How about this simplification?

```ruby
case exitstatus
when true then 0
when false then 1
else Integer(exitstatus)
end
```

File: spec/aruba/processes/in_process_spec.rb [hunk 2]
```diff
@@ -28,11 +29,13 @@ def execute!
     end
   end
 
-  subject(:process) do
```
Comment (mvz) [inline spec/aruba/processes/in_process_spec.rb]: I think you can keep this `subject` but just create a different `let` for `main_class` in the `#exit_status` describe block.

File: spec/aruba/processes/in_process_spec.rb [hunk 3]
```diff
@@ -103,4 +106,30 @@ def execute!
...
+
+  describe "#exit_status" do
+    def run_process(&block)
+      process = build_process Class.new(Runner) { define_method(:execute!, &block) }
+      process.start
+      process.stop
+      process
+    end
+
+    it "exists success" do
```
Comment (mvz) [inline spec/aruba/processes/in_process_spec.rb]: exists -> exits

**Comment by grosser** [comment]
fixed up!

**Comment by grosser** [comment]
Then merge it 😄

**Comment by aslakhellesoy** [comment]
Hi @grosser,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Comment by olleolleolle** [comment]
Yes! And done!

:hearts: :yellow_heart: :green_heart: :blue_heart: :purple_heart:

**Comment by grosser** [comment]
thx for the invite, I'm not really a cucumber user though, just hunting
down a rabbit hole I found :D

**Pull request merged**